WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -243,11 +254,35 @@ final class ProductImageUploader: ProductImageUploaderProtocol {
     func reset() {
         statusUpdatesExcludedProductKeys = []
         statusUpdatesSubscriptions = []
+        backgroundSubscriptions = []
         activeUploadsPublisher = []
 
         actionHandlersByProduct = [:]
         imagesSaverByProduct = [:]
     }
+
+    private func scheduleUploadInProgressNotification() {
+        guard !activeUploadsPublisher.isEmpty else { return }
+
+        let notification = LocalNotification(scenario: .productImageBackgroundUpload)
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+        Task {
+            await LocalNotificationScheduler(pushNotesManager: ServiceLocator.pushNotesManager).schedule(notification: notification,
+                                                                                                         trigger: trigger, remoteFeatureFlag: nil)
+        }
+    }
+
+    @objc private func appDidEnterBackground() {
+        activeUploads
+            .first()
+            .sink { [weak self] activeKeys in
+                guard let self = self else { return }
+                if !activeKeys.isEmpty {
+                    self.scheduleUploadInProgressNotification()
+                }
+            }
+            .store(in: &backgroundSubscriptions)
```

Comment:
Any reason why we need to observe the first value of `activeUploads` instead of triggering `scheduleUploadInProgressNotification` immediately?